### PR TITLE
Cluster index zc

### PR DESCRIPTION
**Pull request by Yetidancer**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR continues the process of setting up curated clusters. It edits models and relationships for the curated clusters, articles, and their joins table. Also creates a show and index view for clusters, and formats the associated controller and methods for appropriate instance variable access. Finally, it sets up tests for the cluster show and index pages. 

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [x] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/curated_clusters_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def index
+    @user = User.find_by(username: params["username"])
+
+    @clusters = @user.curated_clusters
+
+    @serialized_clusters = @clusters.map { |cluster| { id: cluster.id, name: cluster.name } }
+
+
+
+    # require "pry"; binding.pry
```
Comment (ERaith) [inline app/controllers/curated_clusters_controller.rb:11]: You say you need pry... but its commented out. :) What does pry mean? Is that like debugger?

File: app/models/article.rb [hunk 2]
```diff
@@ -18,6 +18,7 @@ class Article < ApplicationRecord
   belongs_to :job_opportunity, optional: true
   belongs_to :organization, optional: true
   belongs_to :collection, optional: true, touch: true
+
```
Comment (ERaith) [inline app/models/article.rb:21]: Did something change on the article model? It seems like maybe this was just an accidental enter.

**Review by holmesm8 — COMMENTED** [review]
I like the set up for the controller and route. Lends itself to a nice curated_cluster index page that's easy to view for the user.

File: app/controllers/curated_clusters_controller.rb [hunk 3]
```diff
@@ -4,13 +4,17 @@ def index
 
     @clusters = user.curated_clusters
 
-    require "pry"; binding.pry
+    @serialized_clusters = @clusters.map { |cluster| { id: cluster.id, name: cluster.name } }
```
Comment (holmesm8) [inline app/controllers/curated_clusters_controller.rb:7]: At this point I think the serializers are potentially unnecessary.

File: spec/features/clusters/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+      cluster_article1 = CuratedClusterArticle.create!(article_id: article1.id, curated_cluster_id: cluster1.id)
+      cluster_article2 = CuratedClusterArticle.create!(article_id: article2.id, curated_cluster_id: cluster1.id)
+      cluster_article3 = CuratedClusterArticle.create!(article_id: article3.id, curated_cluster_id: cluster1.id)
+      cluster_article4 = CuratedClusterArticle.create!(article_id: article4.id, curated_cluster_id: cluster2.id)
+      cluster_article5 = CuratedClusterArticle.create!(article_id: article5.id, curated_cluster_id: cluster3.id)
+      cluster_article6 = CuratedClusterArticle.create!(article_id: article6.id, curated_cluster_id: cluster3.id)
+      require "pry"; binding.pry
+
+      visit "/users/#{user.username}/curated_clusters/#{cluster1.id}"
```
Comment (ERaith) [inline spec/features/clusters/show_spec.rb:27]: Good to know that #{ } is kinda like ${} in JS. In JS though I need to use `Some text ${variable}` and cannot use " double-quotes.

File: app/controllers/curated_clusters_controller.rb [hunk 5]
```diff
@@ -4,13 +4,17 @@ def index
 
     @clusters = user.curated_clusters
 
-    require "pry"; binding.pry
```
Comment (holmesm8) [inline app/controllers/curated_clusters_controller.rb]: Need to take out the prys on the next run through

File: spec/features/clusters/show_spec.rb (hunk 4 above)
Comment (holmesm8) [reply spec/features/clusters/show_spec.rb:27]: Definitely similar syntax. Ruby will require the double quotes when doing interpolation.

**Review by holmesm8 — APPROVED** [review]

**Review by holmesm8 — APPROVED** [review]

File: app/controllers/curated_clusters_controller.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class CuratedClustersController < ApplicationController
+
+  def index
+    user = User.find_by(username: params["username"])
+
+    @clusters = user.curated_clusters
+    
+    require "pry"; binding.pry
```
Comment (holmesm8) [inline app/controllers/curated_clusters_controller.rb]: Take out the pry

**Review by holmesm8 — APPROVED** [review]

**Pull request merged**